File: SampleSqlLite/app/src/main/java/com/sqllite/samplesqllite/NewAnimal.java
package com.sqllite.samplesqllite;

import java.util.HashMap;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;


public class NewAnimal extends Activity {
    EditText animalName;
    EditText animalTallness;

    DBController controller = new DBController(this);

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.add_new_animal);
        animalName = findViewById(R.id.animalName);
        animalTallness = findViewById(R.id.animalTallness);

    }

    public void addNewAnimal(View view) {
        HashMap<String, String> queryValues = new HashMap<>();
        queryValues.put("animalName", animalName.getText().toString());
        queryValues.put("animalTallness", animalTallness.getText().toString());

        controller.insertAnimal(queryValues);
        this.callHomeActivity(view);
    }

    public void callHomeActivity(View view) {
        Intent objIntent = new Intent(getApplicationContext(), MainActivity.class);
        startActivity(objIntent);
    }
}

File: kotlin/SampleSqlLite/app/src/main/java/com/dstrube/samplesqllite/NewAnimal.kt
package com.dstrube.samplesqllite

import android.content.Intent
import android.widget.EditText
import android.os.Bundle
import android.app.Activity
import android.view.View


class NewAnimal : Activity() {
    private var animalName: EditText? = null
    private var animalTallness: EditText? = null

    private var controller = DBController(this)

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.add_new_animal)
        animalName = findViewById(R.id.animalName)
        animalTallness = findViewById(R.id.animalTallness)

    }

    fun addNewAnimal(view: View) {
        val queryValues = HashMap<String, String>()
        queryValues["animalName"] = animalName?.text.toString()
        queryValues["animalTallness"] = animalTallness?.text.toString()

        controller.insertAnimal(queryValues)
        this.callHomeActivity(view)
    }

    private fun callHomeActivity(view: View) {
        val objIntent = Intent(applicationContext, MainActivity::class.java)
        startActivity(objIntent)
    }
}
